sled-agent/src/bootstrap/early_networking.rs

```diff
@@ -443,38 +445,36 @@ impl<'a> EarlyNetworkSetup<'a> {
                 "config" => #?dpd_port_settings
             );
 
-            loop {
-                match dpd
-                    .port_settings_apply(
-                        &port_id,
-                        Some(OMICRON_DPD_TAG),
-                        &dpd_port_settings,
-                    )
-                    .await
-                {
-                    Ok(_) => break Ok(()),
-                    Err(e) => {
-                        if let Some(StatusCode::SERVICE_UNAVAILABLE) =
-                            e.status()
-                        {
-                            warn!(
-                                self.log,
-                                "unable to apply uplink configuration, dendrite not available";
-                                "port_id" => ?port_id,
-                                "configuration" => ?dpd_port_settings,
-                            );
-                            sleep(Duration::from_secs(5)).await;
-                            continue;
-                        } else {
-                            break Err(EarlyNetworkSetupError::Dendrite(
-                                format!(
-                                "unable to apply uplink port configuration: {e}"
-                            ),
-                            ));
-                        }
-                    }
+            while let Err(e) = dpd
+                .port_settings_apply(
+                    &port_id,
+                    Some(OMICRON_DPD_TAG),
+                    &dpd_port_settings,
+                )
+                .await
+            {
+                if let Some(StatusCode::SERVICE_UNAVAILABLE) = e.status() {
+                    warn!(
+                        self.log,
+                        "dendrite not available, re-attempting port configuration in 5 seconds";
+                        "port_id" => ?port_id,
+                        "configuration" => ?dpd_port_settings,
+                    );
+                    sleep(Duration::from_secs(5)).await;
+                    continue;
+                } else {
+                    // log and move on to the next uplink instead of bailing on the
+                    // entire uplink process
+                    error!(
```

Comment:
It should probably still be an error if all of the uplinks fail.

Reply:
Good idea. Maybe maintain a count in the outer loop.

Reply:
done